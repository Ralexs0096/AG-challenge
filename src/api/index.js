import axios from 'axios'

export const baseUrl = 'https://covid19.mathdro.id/api'

export const fetchData = async country => {
  try {
    const countryData = `${baseUrl}/countries/${country}`
    const { data } = await axios.get(countryData)

    const { confirmed, deaths, recovered, lastUpdate } = data

    const lastDateUpdated = lastUpdate.split('T')[0]

    return { confirmed, deaths, recovered, lastDateUpdated }
  } catch (error) {
    return error
  }
}
