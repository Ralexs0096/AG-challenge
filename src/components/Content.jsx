import PropTypes from 'prop-types'
import Loader from './Loader'

const Content = ({ data, country }) => {
  if (!data) {
    return <Loader />
  }

  const { confirmed, deaths, recovered, lastDateUpdated } = data

  return (
    <div className="container">
      <h2>
        Results for: <span className="title-header-country">{country}</span>
      </h2>
      <div className="row">
        <div className="card card-1 card-cm">
          <h3>Confirmed Cases</h3>
          <span className="title-header">
            {confirmed.value.toLocaleString()}
          </span>
        </div>

        <div className="card card-2 card-cm">
          <h3>Death Toll</h3>
          <span className="title-header">{deaths.value.toLocaleString()}</span>
        </div>
        <div className="card card-3 card-cm">
          <h3>Recovered</h3>
          <span className="title-header">{recovered.value}</span>
        </div>
        <div className="card card-4 card-cm">
          <h3>Last Updated</h3>
          <span className="title-header">{lastDateUpdated}</span>
        </div>
      </div>
    </div>
  )
}

Content.propTypes = {
  country: PropTypes.string.isRequired,
  data: PropTypes.object,
}

export default Content
