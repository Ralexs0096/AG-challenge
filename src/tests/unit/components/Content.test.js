import React from 'react'
import { shallow } from 'enzyme'
import Content from '../../../components/Content'
import { fetchData } from '../../../api'
jest.mock('../../../api')

describe('Content test', () => {
  const country = 'nicaragua'
  const data = {
    confirmed: { value: 18004 },
    recovered: { value: 0 },
    deaths: { value: 224 },
    lastUpdate: '2022-02-27T22:21:05.000Z',
  }

  const wrapper = shallow(<Content data={data} country={country} />)

  test('Should show Content component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should display the data upload items', () => {
    fetchData.mockReturnValue({ data })
    expect(wrapper).toMatchSnapshot()
  })

  test('should not display the Loader component', () => {
    expect(wrapper.find('Loader').exists()).not.toBe(true)
  })

  test('should display the Country prop within the H2 tag', () => {
    expect(wrapper.find('span').at(0).text()).toBe(country)
  })
})
