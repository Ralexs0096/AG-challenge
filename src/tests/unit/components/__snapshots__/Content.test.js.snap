// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content test Should show Content component 1`] = `
<div
  className="container"
>
  <h2>
    Results for: 
    <span
      className="title-header-country"
    >
      nicaragua
    </span>
  </h2>
  <div
    className="row"
  >
    <div
      className="card card-1 card-cm"
    >
      <h3>
        Confirmed Cases
      </h3>
      <span
        className="title-header"
      >
        18,004
      </span>
    </div>
    <div
      className="card card-2 card-cm"
    >
      <h3>
        Death Toll
      </h3>
      <span
        className="title-header"
      >
        224
      </span>
    </div>
    <div
      className="card card-3 card-cm"
    >
      <h3>
        Recovered
      </h3>
      <span
        className="title-header"
      >
        0
      </span>
    </div>
    <div
      className="card card-4 card-cm"
    >
      <h3>
        Last Updated
      </h3>
      <span
        className="title-header"
      />
    </div>
  </div>
</div>
`;

exports[`Content test should display the data upload items 1`] = `
<div
  className="container"
>
  <h2>
    Results for: 
    <span
      className="title-header-country"
    >
      nicaragua
    </span>
  </h2>
  <div
    className="row"
  >
    <div
      className="card card-1 card-cm"
    >
      <h3>
        Confirmed Cases
      </h3>
      <span
        className="title-header"
      >
        18,004
      </span>
    </div>
    <div
      className="card card-2 card-cm"
    >
      <h3>
        Death Toll
      </h3>
      <span
        className="title-header"
      >
        224
      </span>
    </div>
    <div
      className="card card-3 card-cm"
    >
      <h3>
        Recovered
      </h3>
      <span
        className="title-header"
      >
        0
      </span>
    </div>
    <div
      className="card card-4 card-cm"
    >
      <h3>
        Last Updated
      </h3>
      <span
        className="title-header"
      />
    </div>
  </div>
</div>
`;
